docs/DeveloperGuide.md

```diff
@@ -2,37 +2,200 @@
+a unique ID, a date of birth and gender. The PatientList class holds an ArrayList of Patients and manipulates them accordingly.
+
+At the start of the program, a new PatientList object is instantiated. Through methods in the Storage class, data is read from the
+relevant text files to create Patients that existed prior to the last closure of the program, and then adds these patients to the
+ArrayList in PatientList. This finishes the initial set-up.
+
+### Methods in PatientList class:
+
+addPatient - this method takes in the aforementioned variables through UI class and parses them. If they are all valid, a new
+Patient is created and added to the ArrayList in PatientList.
+
+findPatient - this method takes in an ID, iterates through the ArrayList of Patients and compares the ID with the ID of each of the
+existing Patients in the list. If there is a match, the corresponding Patient is returned. Else, returns null.
+
+retrievePatient - this method is very similar to findPatient, but instead of returning the Patient found, it prints out the details
+of the Patient using the toString method in the Patient class.
+
+listPatients - this method first checks if the ArrayList of Patients is non-empty. If not so, it prints a message that there are
+no patients in the system currently and terminates.
+Else, it iterates through the ArrayList of Patients and uses the toString method in Patient to print out the details of each Patient.
+
+modifyPatientDetails - this method takes in name, birthDate, gender and ID. It tries to find the patient with a matching ID.
+If the patient is not found, returns. Else, if the name is not an empty String, replace the existing name with the input name.
+Repeat for birthDate and gender.
+
+### VisitList Component
+The `VisitList` Component,
+* stores the list of visits
+* can add new visit to the list
+* can edit reason for an existing visit in the list
+* can view all visits in the list
+* can view all visits for a patient
+* can view a specific visit
+* depends on `UI` class (as the `VisitList` component interacts with user through the UI component, and makes use of its methods to print details)
+
+**Methods in `VisitList` class:**
+* `addVisit` - This method allows user to add a visit to the `VisitList` by specifying `id` of patient, `dateOfVisit`, and `timeOfVisit`.
+* `editReason` - This method allows user to edit reason for an existing visit, by specifying `id` of patient and `reason` for visit
+* `viewAll` - This method iterates through the list of all visits, and prints each visit record
+* `viewPatient` - This method iterates through the list of all visits, and prints the visit records that match the specified `id` of patient
+* `viewVisit` - This method iterates through the list of all visits, and prints the visit record that matches the specified `index` of the visit
+
+### PrescriptionList component
+**API**: `PrescriptionList.java`
+
+![](images/PrescriptionListClassDiagram.png)
```

Comment:
Remember to put the type of commands at the end of the diagram (e.g. viewPatientPrescription(ui: UI, patientId: String)): void